components/Modal/Schedule/index.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from "react";
+
+import Image from "next/image";
+
+import classNames from "classnames/bind";
+import dayjs from "dayjs";
+import { useShallow } from "zustand/react/shallow";
+
+import Input from "@/components/Input";
+import { useDateStore } from "@/store/useDateStore";
+import { useModalStore } from "@/store/useModalStore";
+
+import styles from "./ScheduleModal.module.scss";
+
+const cn = classNames.bind(styles);
+
+interface ScheduleModalProps {
+  mode: "create" | "edit" | "detail";
+}
+
+export default function ScheduleModal({ mode }: ScheduleModalProps) {
+  const [isReadOnly, setIsReadOnly] = useState<boolean>(mode === "detail");
+  const { closeModal } = useModalStore();
+  const { scheduleStart, scheduleEnd, setScheduleStart, setScheduleEnd } = useDateStore(
+    useShallow((state) => ({
+      scheduleStart: state.scheduleStart,
+      scheduleEnd: state.scheduleEnd,
+      setScheduleStart: state.setScheduleStart,
+      setScheduleEnd: state.setScheduleEnd,
+    })),
+  );
+
+  const handleCloseModal = () => {
+    closeModal();
+    setScheduleStart(null);
+    setScheduleEnd(null);
+  };
+
+  const handleSubmit = (mode: "create" | "edit") => {
+    //create 이면 post, edit이면 put 요청 추가
+    handleCloseModal();
+  };
+
+  return (
+    <div className={cn("container")}>
+      <form className={cn("create-schedule-form")}>
+        <input
+          type="text"
+          className={cn("schedule-name")}
+          readOnly={isReadOnly}
```

Comment:
```suggestion
          readOnly={mode === "detail"}
```
p4: `mode === "detail"`일 때만 사용하는 것 같은데 직접 넣지 않고 따로 상태로 빼신 이유가 있나요??

Reply:
아직 모달은 구현 안된상태이긴한데 일단 제가 계획한 느낌은
모달을 하나로해서 생성, 상세, 수정을 mode 상태에때라서 보여주려고했던거같아요
상태에 초기값으로 mode prop 넣어주는식으로 안하면 mode가 바꼇을때 처리를 못할거같아서 일단이렇게 했습니다! 
아직 구현 초기라서 수정될수도있어요

Reply:
넵 무슨 말씀인지 이해했습니다!
아직 구현이 안되서 정확히는 모르겠지만 뭔가 구현이 됐을 때 만능 컴포넌트(상세, 수정, 생성이 다 가능한)이기 때문에 조금 무겁고 추가사항이 늘어났을 때 유지보수하기 힘들 수도 있을 것 같다는 생각이 들어서 `ScheduleModal`컴포넌트는 UI 로직만 가지고 `ScheduleEditModal`, `ScheduleCreateModal`컴포넌트를 만들고 `ScheduleModal`컴포넌트를 `import`해서 UI로직이랑 비지니스 로직을 나눠봐도 괜찮을 것 같다는 생각이 들어요!(이건 제 개인적인 생각입니다🧐 참고만 해주세요🙃)
- `ScheduleModal` 컴포넌트(UI)
```jsx
export default function ScheduleModal({ onClick }: ScheduleModalProps) {
    // 일정 모달의 UI 로직
    <button type="submit" onClick={onClick}>
}
```
- `ScheduleEditModal` 컴포넌트(비지니스 로직)
```jsx
export default function ScheduleEditModal() {
    const handleEditButtonClick = async () => {
        // 일정 수정 로직
    }
    
    return <ScheduleModal onClick={handleEditButtonClick} />
}
```